Fixed the list to display entries alphabetically

Perseus opened the pull request:
Fixes issue #966 

Removed the conditional check for `$list_id` hence displaying the list items, which are retrieved alphabetically from the database, in order.

Perseus commented:
I'm also working on the type-ahead selector. Wanted to know how you guys wanted it done. A separate text field which the user can type in to find a list or something else?

Perseus commented:
@tmccormi

teryhill commented:
Fails testing.

Perseus commented:
Is the top option still "language"? What is the issue?

teryhill commented:
With the change you get this

![image](https://user-images.githubusercontent.com/4655366/35736925-c70866a4-07f7-11e8-9bcf-b43f2d8adc7b.png)

The option changes in the drop down but the listed items do not change. Then if you select a list you get this.

![image](https://user-images.githubusercontent.com/4655366/35737017-18fc6b90-07f8-11e8-8b0f-d4eab3205bd1.png)

Which shows the list but the wrong header.

teryhill commented:
This is what I came up with.

![image](https://user-images.githubusercontent.com/4655366/35737574-0198eea4-07fa-11e8-98f1-a9e2a092a45d.png)

Then when you select a list you get the list and the save button.

![image](https://user-images.githubusercontent.com/4655366/35737626-293696b4-07fa-11e8-86d0-140b4d9a8e89.png)

Perseus commented:
@teryhill I changed it to be blank initially and nothing displays and then can be selected as needed. 
![2438bc9e03fc3f0e1ce0f864c8ccc03f](https://user-images.githubusercontent.com/6716458/35763523-0a046156-08d4-11e8-9039-e70acb9d8162.png)
![dbed3daef352c64399db5a38f4f83f67](https://user-images.githubusercontent.com/6716458/35763524-0c2ea306-08d4-11e8-9422-30a3f714e0d3.png)
![f6438d0248f62a2a29c6e32cbae7dbfb](https://user-images.githubusercontent.com/6716458/35763526-0d15ba52-08d4-11e8-82eb-a2a003beeb75.png)
Does this look fine?

teryhill commented:
Yes it does look fine.

teryhill commented on interface/super/edit_list.php:
```diff
@@ -858,7 +861,7 @@ function mysubmit() {
 
 <center>
-
+<?php if ($list_id != ' '): ?>
```
Should use {

Perseus replied:
Alright I'll fix that.

teryhill commented on interface/super/edit_list.php:
```diff
@@ -1046,7 +1049,7 @@ function mysubmit() {
 
 </form>
-
+<?php endif; ?>
```
Should use }

aethelwulffe commented:
OK, I just went and tested an out-of-date commit...Lemme try again.

Perseus commented:
@aethelwulffe Are there any future plans to upgrade the jQuery versions being used? The edit list page uses 1.2 and most of the libraries having type-ahead selectors don't really support that. I could work on something just for that but it'd be easier if jQuery was updated instead. Less legacy code to maintain

teryhill commented:
@Perseus it is an on going project. Issue 416 has the info on it.

Perseus commented:
Please review the commit @teryhill @aethelwulffe . I couldn't find any errors with the updated jquery version in the edit_list page and the list seems to work properly. Please let me know if any changes are needed.

aethelwulffe commented on interface/super/edit_list.php:
```diff
@@ -618,9 +618,11 @@ function writeITLine($it_array) {
 <!-- supporting javascript code -->
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery.js"></script>
-
+<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery-3.3.1.min.js"></script>
```
`require_once $GLOBALS['srcdir'].'/headers.inc.php'; ` is on line 22 of this file.  

To call a single library from the assets/js/ directory, use `function include_js_library($path)` where $path is the directory name of the asset under assets/js/ and the actual file name is index.js.

`function include_css_library($path)` is the same as the above, just requires the full file path.

`call_required_libraries($library_array)` allows multiple libraries to be called.  In your case, you would invoke it with `call_required_libraries('jquery-min-3-3-1','select2');`

The above requires renaming the files to index.js, and requires that the directory naming standard be put into the jquery-min-* naming standard.  Either a default case switch or a final elseif else statement needs to be added that invokes whatever path is passed in the argument array, OR add "select2" to the list of 'if' conditions.

These functions require some rework but when we change the functions themselves to make them more efficient (lots of people touched the file) and provide default vs. specified behavior of include_js_library($path) to be added to call_required_libraries , but we should not have to ever change the calls.

Perseus commented:
@aethelwulffe I've moved the jquery file to the assets directory,  added a select2 clause to the headers file and added the required libraries function call to the edit list page

aethelwulffe commented:
Yep, all the pieces work.
